File: mmap_demo/app/src/main/java/com/example/mmap_demo/MainActivity.java
package com.example.mmap_demo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.example.mmap_demo.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    // Used to load the 'mmap_demo' library on application startup.
    static {
        System.loadLibrary("mmap_demo");
    }

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // Example of a call to a native method

        initEvent();
    }

    private void initEvent() {
        binding.btnTestMmapOne.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                testMMAp();
            }
        });


    }

    /**
     * A native method that is implemented by the 'mmap_demo' native library,
     * which is packaged with this application.
     */
    public native String testMMAp();
}

File: TestCPU/app/src/main/java/com/jar/testcpu/MainActivity.kt
package com.jar.testcpu

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.ArrayMap
import android.widget.TextView
import com.jar.testcpu.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // Example of a call to a native method
        binding.sampleText.text = stringFromJNI()
    }

    /**
     * A native method that is implemented by the 'testcpu' native library,
     * which is packaged with this application.
     */
    external fun stringFromJNI(): String

    companion object {
        // Used to load the 'testcpu' library on application startup.
        init {
            System.loadLibrary("testcpu")
        }
    }
}